storage/src/mmr/mutable.rs

```diff
@@ -242,10 +283,61 @@ impl<K: Array, V: Array, H: CHasher> MutableMmr<K, V, H> {
         proof.verify_range_inclusion(hasher, &digests, start_pos, end_pos, root_hash)
     }
 
-    /// Consume the store and return its log of operations, from which the state of the store can be
-    /// recovered.
-    pub fn to_log(self) -> Vec<Operation<K, V>> {
-        self.log
+    /// Consume the store and return a [StoreState] from which the current active state of the store
+    /// can be fully recovered and fully proven.
+    pub fn to_state(mut self) -> StoreState<K, V, H> {
+        self.prune_known_inactive();
+
+        let prune_to_pos = leaf_num_to_pos(self.pruned_loc);
+        let pinned_nodes = self.ops.node_digests_to_pin(prune_to_pos);
+
+        StoreState {
+            log: self.log.into(),
+            pruned_loc: self.pruned_loc,
+            pinned_nodes,
+        }
+    }
+
+    /// Raise the inactivity floor as high as we can without moving more than `max_moves` ops to
+    /// tip. This method does not change the state of the store's snapshot, but it does change the
+    /// root hash of the store because of the moved operations.
+    fn raise_inactivity_floor(&mut self, hasher: &mut H, mut max_moves: usize) {
```

Comment:
I think we should consider each iteration as a move, regardless of whether we rewrite (otherwise we may iterate over a TON of data at once).

The algorithm should strive to make update/compaction symmetric IMO. The nice byproduct of this, is that the code should get even simpler.

Reply:
changed so each "step" is a any advancement of the floor whether or not it involves a move.